web/source/osk/visualKeyboard.ts

```diff
@@ -786,9 +786,9 @@ namespace com.keyman.osk {
           // All key widths and paddings are rounded for uniformity
           var keyPercent: number, padPercent: number, totalPercent=0;
           for(j=0; j<keys.length-1; j++) {
-            keyPercent = Math.round(keys[j]['widthpc'] * objectWidth); //Math.round(parseInt(keys[j]['width'],10)*objectWidth/totalWidth);
+            keyPercent = keys[j]['widthpc'] * objectWidth;
             keys[j]['widthpc']=keyPercent;
-            padPercent = Math.round(keys[j]['padpc'] * objectWidth); // Math.round(parseInt(keys[j]['pad'],10)*objectWidth/totalWidth);
+            padPercent = keys[j]['padpc'] * objectWidth;
```

Comment:
Ah! So was this a case of premature rounding?

Reply:
Indeed, rounding here there and everywhere. We don't actually need to round at all -- this was due to limitations in, from memory??, Android Browser, which only accepted integer CSS values for width.